File: Interviews/LeetCode/Daily-Problems/2022/November/Day-13/151.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words, word = [], ""
        for c in s:
            if c != ' ':
                word += c
            else:
                if len(word):
                    words.append(word)
                    word = ""
        if len(word):
            words.append(word)
        ans = ""
        for idx, c in enumerate(reversed(words)):
            ans += c
            if idx < len(words) - 1:
                ans += ' '

        return ans

File: Interviews/LeetCode/Daily-Problems/2022/November/Day-13/151.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words;
        s += ' ';
        string word = "";
        for (char c: s) {
            if (c != ' ') {
                word += c;
            } else {
                if (word.size() > 0) {
                    words.push_back(word);
                    word = "";
                }
            }
        }
        
        string ans = "";
        for (int i = words.size() - 1; i >= 0; i--) {
            ans += words[i];
            if (i > 0) {
                ans += ' ';
            }
        }
        
        return ans;
    }
};